vespalib/src/vespa/vespalib/util/shared_string_repo.h

```diff
@@ -34,21 +35,43 @@ class SharedStringRepo {
 
     class alignas(64) Partition {
     public:
-        struct Entry {
+        class Entry {
+        public:
             static constexpr uint32_t npos = -1;
-            uint32_t hash;
-            uint32_t ref_cnt;
-            vespalib::string str;
-            explicit Entry(uint32_t next) noexcept : hash(), ref_cnt(next), str() {}
+        private:
+            uint32_t _hash;
+            uint32_t _ref_cnt;
+            vespalib::string _str;
+        public:
+            explicit Entry(uint32_t next) noexcept
+                : _hash(next), _ref_cnt(npos), _str() {}
+            constexpr uint32_t hash() const noexcept { return _hash; }
+            constexpr const vespalib::string &str() const noexcept { return _str; }
+            constexpr bool is_free() const noexcept { return (_ref_cnt == npos); }
             uint32_t init(const AltKey &key) {
-                uint32_t next = ref_cnt;
-                hash = key.hash;
-                ref_cnt = 1;
-                str = key.str;
+                uint32_t next = _hash;
+                _hash = key.hash;
+                _ref_cnt = 1;
+                _str = key.str;
                 return next;
             }
             void fini(uint32_t next) {
-                ref_cnt = next;
+                _hash = next;
```

Comment:
The name of this function is not obvious to me.

Reply:
init/fini is like ctor/dtor